console/src/tasks.rs

```diff
@@ -362,6 +355,14 @@ impl Task {
             .idle
             .unwrap_or_else(|| self.total(since) - self.busy())
     }
+
+    fn updated(&mut self) {
```

Comment:
very nitpicky, but, since this mutates the task in place, i would probably call it
```suggestion
    fn update(&mut self) {
```
i would probably expect something called `updated` to take a `Task` by value and return it.

Reply:
also a nit, but it looks like this currently only handles marking tasks as completed. my guess is that the intention is for it to also eventually do other task state updates, but if that's not the case, it might be worth giving it a name that's specific to completion?